content/kubermatic/master/guides/custom_ca/_index.en.md

```diff
@@ -0,0 +1,123 @@
++++
+title = "Using Custom CAs"
+date = 2021-02-10T11:07:15+02:00
+weight = 10
++++
+
+This document describes how to use a custom set of CA certificates in Kubermatic Kubernetes
+Platform (KKP). A general understanding of TLS certificates is assumed.
+
+KKP 2.17+ allows to configure a "CA bundle" (a set of CA certificates) on the master cluster.
+This CA bundle is then automatically
+
+* copied into each seed cluster
+* copied into each usercluster namespace
+* copied into each usercluster (into the `kube-system` namespace)
+* used for various components, like the KKP API, machine-controller, kube-apiserver etc.
+
+Changes to the CA bundle are automatically reconciled across these locations. If the CA bundle
+is invalid, no further reconciliation happens, so if the master cluster's CA bundle breaks,
+seed clusters are not affected.
+
+Do note that the CA bundle configured in KKP is usually _the only_ source of CA certificates
+for all of these components, meaning that no certificates are mounted from any of the Seed
+cluster host systems.
+
+## Configuration
+
+The CA bundle is stored as a single ConfigMap in the `kubermatic` namespace on the master cluster.
+It needs to have a `ca-bundle.pem` key, which is simply the collection of all PEM-encoded CA
+certificates.
+A good source for a general purpose CA bundle is Mozilla's CA database. The curl maintainers
+provide a [ready-to-use bundle file](https://curl.se/docs/caextract.html) that can be used as
+a starting point and then extended with your own CAs.
+
+KKP ships the CA bundle mentioned above by default in the `kubermatic-operator` Helm chart.
+This means that after installing KKP, a usable default CA bundle is available right from the
+start.
+
+To override the CA bundle, either overwrite the `static/ca-bundle.pem` file in the
+`kubermatic-operator` Helm chart with your own and then `helm upgrade` the operator, or manually
+change the ConfigMap later using `kubectl edit`. Changes to the ConfigMap are picked up
+automatically and are then reconciled.
```

Comment:
If someone changes the ConfigMap and then runs `helm upgrade`, is Helm going to override the CA bundle?

Reply:
I am not sure, but I hope nobody will just flipflop between different ways of managing their CA bundle. :-D